content/blog/2024-10-21-zenoh-firesong.md

```diff
@@ -0,0 +1,936 @@
+        .wait()
+        .unwrap();
+
+    // OPTION: The comprehensive allocation policy that deallocates up to 1000 buffers if provider is not able to allocate
+    let _comprehensive_alloc = buffer_layout
+        .alloc()
+        .with_policy::<Deallocate<1000, Defragment<GarbageCollect>>>()
+        .wait()
+        .unwrap();
+
+    default_alloc
+}
+.await;
+```
+
+For more information, please see the [migration guide](https://zenoh.io/docs/migration_1.0/rust/#shared-memory).
+
+---
+
+## Plugins
+
+In Zenoh 1.0.0 we finished porting the Zenoh ecosystem from async-std to Tokio. All plugins and storage backends now use Tokio.
+
+As a reminder, Zenoh 0.11.0 added the ability for user-applications to load compiled plugins written in Rust, regardless of which language bindings you are using. See the configuration example on how to load the plugins.
+
+When loading a plugin, it must have been built with the same version of the Rust compiler as the bindings loading it, and the `Cargo.lock` of the plugin must be synced with the same commit of Zenoh.
+
+This means that if the language bindings are using `rustc` version `1.75`:
+- The plugin must be built with the same toolchain version `1.75`.
+- The plugin must be built with the same Zenoh Commit.
+- The plugin `Cargo.lock`  had its packages synced with the Zenoh `Cargo.lock`. 
+
+The reason behind this strict set of requirements is due to Rust making no guarantees regarding data layout in memory. This means between compiler versions, the representation may change based on optimizations. More on this topic here: [Rust:Type-Layout](https://doc.rust-lang.org/reference/type-layout.html#representations)
+
+---
+
+## Interest protocol
+
+Zenoh 1.0.0 introduces a major change in the routing protocol that brings a great scalability improvement and a major discovery traffic consumption reduction.
+
+Up to version 0.11.0, all subscribers, queryables and liveliness tokens declarations were propagated to all nodes of the system. 
+In a large system, this could be problematic for small devices. In version 1.0.0, subscribers, queryables and liveliness tokens declarations are not propagated any more to clients and peers subsystems. 
+This implies that writer side filtering cannot be performed any more in clients and peers subsystems. 
+All publications, queries and liveliness queries are sent to the nearest router. 
+Writer side filtering can be reactivated for publications on some key expressions by simply declaring a *publisher*.
+
+---
+
+## Access Control
+
+For Zenoh 1.0.0, we are happy to share that we have added support for TLS and user/password authentication methods as means to identify access control subjects, as relying solely on interface names to identify subjects quickly reaches its limits. 
+This addition introduced the need for a way to describe subjects as combinations of multiple attributes of different types (interface, certificate, common name and/or username). 
```

Comment:
certificate and common name are not seperate attributes, it's one attribute called "certificate common name".
```suggestion
This addition introduced the need for a way to describe subjects as combinations of multiple attributes of different types (interface, certificate common name, and/or username). 
```